### PR TITLE
Prepare auth-next for merging to master

### PR DESCRIPTION
**Pull request by Feiyang1**

### CODE REVIEW

File: packages-exp/auth-compat-exp/src/recaptcha_verifier.ts [hunk 1]
```diff
@@ -33,6 +33,10 @@ export class RecaptchaVerifier
     impl.assertFn(app.options?.apiKey, impl.AuthErrorCode.INVALID_API_KEY, {
       appName: app.name
     });
-    super(container, parameters, (app.auth!() as unknown) as externs.Auth);
+    super(
+      container,
+      parameters as any,
```
Comment (Feiyang1) [inline packages-exp/auth-compat-exp/src/recaptcha_verifier.ts:38]: I cast it to `any` to workaround the TS error - 
```
Argument of type 'object | null | undefined' is not assignable to parameter of type 'Parameters'.
  Type 'undefined' is not assignable to type 'Parameters'.
```
The error seems legit. Is the casting safe here?

File: packages-exp/auth-exp/test/integration/flows/phone.test.ts [hunk 2]
```diff
@@ -69,7 +70,11 @@ describe('Integration test: phone auth', () => {
     auth = getTestInstance();
     fakeRecaptchaContainer = document.createElement('div');
     document.body.appendChild(fakeRecaptchaContainer);
-    verifier = new RecaptchaVerifier(fakeRecaptchaContainer, undefined, auth);
+    verifier = new RecaptchaVerifier(
+      fakeRecaptchaContainer,
+      undefined as any,
```
Comment (Feiyang1) [inline packages-exp/auth-exp/test/integration/flows/phone.test.ts:75]: Has to cast to `any` to workaround TS error here and below. `RecaptchaVerifier` doesn't take `undefined` as the second argument, but I guess it's okay if the test doesn't touch the second argument.

**Comment by Feiyang1** [comment]
Renamed `recaptcha.d.ts` to `recaptcha.ts`, otherwise `recpatcha.d.ts` is not created in the `/dist` folder causing broken typings. We should always use `.ts` files in the implementation package, even just for defining types.

File: packages-exp/auth-compat-exp/src/recaptcha_verifier.ts (hunk 1 above)
Comment (samhorlbeck) [reply packages-exp/auth-compat-exp/src/recaptcha_verifier.ts:39]: Yeah the typing in the parent class is wrong. I'll need to go in and fix it there

**Review by samhorlbeck — APPROVED** [review]
Thanks so much for doing this Fei!

**Review by avolkovi — APPROVED** [review]

**Pull request merged**